serveradmin/common/static/js/serveradmin.js

```diff
@@ -7,23 +7,58 @@ $(document).ready(function() {
     // This is our progress spinner we can call e.g. spinner.enable() from
     // everywhere when ever we need it for example when doing long running
     // ajax requests for the servershell search.
-    spinner = {
+    window.spinner = {
+        _id: null,
+        _running: 0,
+        _spinner: $('#spinner'),
+        _counter: $('#spinner-counter'),
+        _value: $('#spinner-counter-value'),
+        _toggle_css: function() {
+            if (this._spinner.css('display') === 'none') {
+                this._spinner.show();
+                this._counter.show();
+            }
+
+            if (this._spinner.hasClass('text-success')) {
+                this._spinner.removeClass('text-success');
+                this._spinner.addClass('text-secondary');
+
+                this._counter.removeClass('text-success');
+                this._counter.addClass('text-secondary');
+
+                this._spinner.css('animation-play-state', 'paused');
+            }
+            else {
+                this._spinner.removeClass('text-secondary');
+                this._spinner.addClass('text-success');
+
+                this._counter.removeClass('text-secondary');
+                this._counter.addClass('text-success');
+
+                this._spinner.css('animation-play-state', 'running');
+            }
+        },
         enable: function () {
-            $('#spinner').show();
-            $('#spinner-counter').show();
-
-            let update_spinner = function() {
-                let element = $('#spinner-counter-time');
-                element.html(Number.parseInt(element.html()) + 1);
-                setTimeout(update_spinner, 1);
-            };
-            setTimeout(update_spinner, 1);
+            this._running++;
+            if (this._running > 1) {
+                return;
+            }
+
+            this._value.html(0);
+            this._toggle_css();
+
+            this._id = setInterval(function () {
+                spinner._value.html(Number.parseInt(spinner._value.html()) + 1);
+            }, 1);
```

Comment:
It looked kind of odd to have it increasing every 100ms and most Servershell Queries takes less then 100ms.

What came to my mind (avoided for know because to much effort) is. Update the timer display every 10 or 50 ms but measure the time by getting setting the diff between a start timestamp and the current timestamp instead of counting. What do you think about that ?

Reply:
Yeah, could be a possibility, and would cost some rewrite, I guess. If 1ms works fine for now for the browser, I don't think that it is worth it. Could be a topic to think about in the future though.